src/main/java/graphql/InvalidSyntaxError.java

```diff
@@ -2,32 +2,37 @@
 
 
 import graphql.language.SourceLocation;
-import org.antlr.v4.runtime.RecognitionException;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class InvalidSyntaxError implements GraphQLError {
+import static java.util.Collections.singletonList;
+
+public class InvalidSyntaxError extends GraphQLException implements GraphQLError {
 
     private final String message;
+    private final String sourcePreview;
     private final List<SourceLocation> locations = new ArrayList<>();
 
     public InvalidSyntaxError(SourceLocation sourceLocation, String msg) {
-        this.message = mkMessage(msg);
-        if (sourceLocation != null) {
-            this.locations.add(sourceLocation);
-        }
+        this(singletonList(sourceLocation), msg);
     }
 
     public InvalidSyntaxError(List<SourceLocation> sourceLocations, String msg) {
+        this(sourceLocations, msg, null, null);
+    }
+
+    public InvalidSyntaxError(List<SourceLocation> sourceLocations, String msg, String sourcePreview, Exception cause) {
+        super(cause);
         this.message = mkMessage(msg);
+        this.sourcePreview = sourcePreview;
         if (sourceLocations != null) {
             this.locations.addAll(sourceLocations);
         }
     }
 
     private String mkMessage(String msg) {
-        return "Invalid Syntax" + (msg == null ? "" : " : " + msg);
+        return "Invalid Syntax" + (msg == null ? "" : " : " + msg + "");
```

Comment:
do we need that empty string at the end?

Reply:
fixed